security_monkey/manage.py

```diff
@@ -665,11 +665,14 @@ def sync_swag(owner, bucket_name, bucket_prefix, bucket_region, account_type, sp
         if s3_name:
             custom_fields['s3_name'] = s3_name
 
-        for service in services:
-            if service['name'] == 's3':
-                c_id = service['metadata'].get('canonicalId')
-                if c_id:
-                    custom_fields['canonical_id'] = c_id
+        s3_service = services_by_name.get('s3', {})
+        if s3_service:
+            c_id = s3_service['metadata'].get('canonicalId', None)
+            if c_id:
+                custom_fields['canonical_id'] = c_id
+        role_name = secmonkey_service.get('metadata', {}).get('role_name', None)
```

Comment:
Dumb question: what happens if `role_name` isn't set?  Will this default to `SecurityMonkey` ?

Reply:
Pretty sure. Security Monkey would likely not work with SWAG at all if it didn't. :smile: